pytato/transform/__init__.py

```diff
@@ -49,8 +49,11 @@
 from pymbolic.mapper.optimize import optimize_mapper
 
 ArrayOrNames = Union[Array, AbstractResultWithNamedArrays]
-MappedT = TypeVar("MappedT", bound=ArrayOrNames)
+MappedT = TypeVar("MappedT",
+                  Array, AbstractResultWithNamedArrays, ArrayOrNames)
 CombineT = TypeVar("CombineT")  # used in CombineMapper
+CopyMapperT = TypeVar("CopyMapperT",  # used in CopyMapper
```

Comment:
```suggestion
MapperResultT = TypeVar("CopyMapperT",  # used in CopyMapper
```
?
Or `CopyMapperResultT`?